core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,611 @@
+    // Kokkos View precondition checks happen in release builds
+    check_basic_view_constructibility(other.mapping());
+
+    static_assert(
+        std::is_constructible_v<data_handle_type,
+                                const typename OtherA::data_handle_type &>,
+        "Kokkos::View: incompatible data_handle_type for View construction");
+    static_assert(std::is_constructible_v<extents_type, OtherE>,
+                  "Kokkos::View: incompatible extents for View construction");
+  }
+
+  KOKKOS_DEFAULTED_FUNCTION constexpr BasicView &operator=(const BasicView &) =
+      default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr BasicView &operator=(BasicView &&) =
+      default;
+
+  // Allocating constructors specific to BasicView
+  ///
+  /// Construct from a given mapping
+  ///
+  explicit constexpr BasicView(const std::string &label,
+                               const mapping_type &mapping)
+      : BasicView(view_alloc(label), mapping) {}
+
+  ///
+  /// Construct from a given extents
+  ///
+  explicit constexpr BasicView(const std::string &label,
+                               const extents_type &ext)
+      : BasicView(view_alloc(label), mapping_type{ext}) {}
+
+ private:
+  template <class... P>
+  data_handle_type create_data_handle(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      const typename mdspan_type::mapping_type &arg_mapping) {
+    constexpr bool has_exec = Impl::ViewCtorProp<P...>::has_execution_space;
+    // Copy the input allocation properties with possibly defaulted properties
+    // We need to split it in two to avoid MSVC compiler errors
+    auto prop_copy_tmp =
+        Impl::with_properties_if_unset(arg_prop, std::string{});
+    auto prop_copy = Impl::with_properties_if_unset(
+        prop_copy_tmp, memory_space{}, execution_space{});
+    using alloc_prop = decltype(prop_copy);
+
+    if (alloc_prop::initialize &&
+        !alloc_prop::execution_space::impl_is_initialized()) {
+      // If initializing view data then
+      // the execution space must be initialized.
+      Kokkos::Impl::throw_runtime_exception(
+          "Constructing View and initializing data with uninitialized "
+          "execution space");
+    }
+    return data_handle_type(Impl::make_shared_allocation_record<ElementType>(
+        arg_mapping.required_span_size(),
+        Impl::get_property<Impl::LabelTag>(prop_copy),
+        Impl::get_property<Impl::MemorySpaceTag>(prop_copy),
+        has_exec ? &Impl::get_property<Impl::ExecutionSpaceTag>(prop_copy)
+                 : nullptr,
+        std::integral_constant<bool, alloc_prop::initialize>(),
+        std::integral_constant<bool, alloc_prop::sequential_host_init>()));
+  }
+
+ public:
+  template <class... P>
+  explicit inline BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
```

Comment:
No requires-clause here?

Reply:
haha yeah, this was copied from View not from mdspan so it didn't originate in C++20 + land :-) I will put a comment in though.

Reply:
fixed